halo2_proofs/src/plonk/lookup/prover.rs

```diff
@@ -614,9 +614,20 @@ fn permute_expression_pair<C: CurveAffine, R: RngCore>(
     }
     assert!(repeated_input_rows.is_empty());
 
-    permuted_input_expression
-        .extend((0..(blinding_factors + 1)).map(|_| C::Scalar::random(&mut rng)));
-    permuted_table_coeffs.extend((0..(blinding_factors + 1)).map(|_| C::Scalar::random(&mut rng)));
+    permuted_input_expression.extend((0..(blinding_factors + 1)).map(|_| {
+        if cfg!(feature = "zero-knowledge") {
+            C::Scalar::random(&mut rng)
+        } else {
+            C::Scalar::zero()
```

Comment:
in fact if zk is disabled, "0..(blinding_factors + 1" becomes empty so "C::Scalar::zero()" never executed.

what about : 

```
#[cfg(feature = "zero-knowledge")]
permuted_table_coeffs.extend((0..(blinding_factors + 1)).map(|_| C::Scalar::random(&mut rng)));
```